### PR TITLE
[BASE-1389] Update documentation

### PR DESCRIPTION
**Pull request by veronikaKochugova**
1. Fix all links (and add more), content tables, references, titles... delete outdated
2. Add more about SDs, Modes, Load Steps, Bundles and extensions, Basic Terms and others...
https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation/doc/design/architecture#21-storage-driver
https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation/doc/design/architecture#1-basic-terms
https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation#3-bundles-and-extenstions
https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation/doc/design/modes
3. Refactor and fix configuration table
https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation/doc/usage/input/configuration
4. Fix info about deployment, ports, k8s, mount files...
https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation/doc/deployment
5. Rstructure and unify all doc pages. All docs are located ONLY in /doc directory and all subdirectories have own README.md.
6. Restructure Main README page. Add Main links.
7. Add "Get strted" guide with simple commands and examples.
https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation/doc/getstarted
8. add images and other small fixes...
----
9. S3 SD doc update + add example with minio/minio
https://github.com/emc-mongoose/mongoose-storage-driver-s3/blob/master/README.md#s3-storage-driver

### CODE REVIEW

File: doc/comparision/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+## 3. Scalability
+|                                                    | Mongoose  | COSBench | LoadRunner | Locust |
+| ---                                                | :---:     | :---:    | :---:      | :---:  |
+|**Horizontal** (Distributed Mode)                 |:heavy_check_mark:|:heavy_check_mark:| TBD |:heavy_check_mark:|
+|**Vertical** (Max sustained concurrency per instance)| 1_048_576 |[1024](http://cosbench.1094679.n5.nabble.com/how-many-connections-users-can-cosbench-create-to-test-one-swift-storage-tp325p326.html)| TBD |[1_000_000](https://locust.io/)|
+
+## 4. Input
+|                  | Mongoose  | COSBench | LoadRunner | Locust |
+| ---              | :---:     | :---:    | :---:      | :---:  |
+|**GUI**           |:x:|:heavy_check_mark:|:heavy_check_mark:|:heavy_check_mark:|
```
Comment (dadlex) [inline doc/comparision/README.md]: Change to :heavy_check_mark: :)

File: src/test/robot/api/remote/data/aggregated_defaults.yaml [hunk 2]
```diff
@@ -297,4 +208,4 @@ aliasing:
   target: "load-step-limit-time"
 - name: "item-naming-offset"
   deprecated: true
-  target: "item-naming-seed"
+  target: "item-naming-seed"
```
Comment (dadlex) [inline src/test/robot/api/remote/data/aggregated_defaults.yaml]: newline?

File: doc/getstarted/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+It's recommended to begin by exploring the 👉 **[basic terms](../design/architecture#1-basic-terms)** 👈 to better understand the documentation.
+
+## Simple deploy
+
+```
+docker run --network host emcmongoose/mongoose-base
+```
+
+☝️ This command will launch mongoose in **standalone mode**, and the **storage mock** will be used as the tested [storage](../design/architecture#1-basic-terms).
```
Comment (dadlex) [inline doc/getstarted/README.md]: Not sure that there is a need in this link. The definition of storage in basic terms is that "storage that's tested", so tested storage here pretty much covers the definition. Just my 2 cents

File: doc/getstarted/README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
+# Get started
```
Comment (dadlex) [inline doc/getstarted/README.md:1]: Consider adding a line that says something like "Mongoose supports CRUD notation, hence the name of the operations". We have that information somewhere, so we can also leave a link to it. It seems reasonable to have it stated again in the "newcomers" section.

File: doc/getstarted/README.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+☝️ This command will perform **"read"** operation for 60 seconds. Mongoose will "read" all items from specified "item-input-path" (for example, for File System (as Storage) path = `/path/to/directory`) and when "items" run out mongoose will "read" them again (`recycle`).
+
+> 📘 More about:
+> * **[Items](../usage/item)**
+> * **[CLI arguments](../usage/input/cli)** and **[configuartion options](../usage/input/configuration)**
+> * **[Operation types](../usage/load/operations/types)**
+
+## Simple scenario
```
Comment (dadlex) [inline doc/getstarted/README.md:45]: Introduction is missed. Like: Mongoose also supports input parameters using 'scenarios' - script files defining a workload.

**Review by dadlex — CHANGES_REQUESTED** [review]

File: ci/kuberenetes/distributed.yaml [hunk 6]
```diff
@@ -1,10 +1,10 @@
+# Service for nodes RMI-communication
```
Comment (dadlex) [inline ci/kuberenetes/distributed.yaml:1]: Why do we have manifests in the ci/kubernetes dir? We don't have any kubernetes related ci, so probably we should relocate these manifests somewhere to /doc/ dir.

File: doc/getstarted/README.md (hunk 5 above)
Comment (dadlex) [reply doc/getstarted/README.md:45]: maybe this should be included: Any [JSR-223](https://en.wikipedia.org/wiki/Scripting_for_the_Java_Platform) compatible |[XML](https://en.wikipedia.org/wiki/XML)|[ANSI C, Java, .Net, JS]. It's from comparison page. Nor sure though

File: doc/deployment/README.md [hunk 7]
```diff
@@ -36,10 +44,21 @@ The base image may be used in the standalone mode:
...
     --network host \
-    emcmongoose/mongoose[-<TYPE>] [\
-    <ARGS>]
+    emcmongoose/mongoose[-<TYPE>] \
+    [<MONGOOSE CLI ARGS>]
 ```
 
+##### Mount files
+
+An example for mounting and using a scenario. Thus, files for input/output, configurations, logs and metrics can be mounted.
```
Comment (dadlex) [inline doc/deployment/README.md:56]: I would strongly encourage adding next info: Mounting a volume is the only right way to save Mongoose logs in docker. Don't get confused by `--output-file` option, it's not about docker. 
Probably this info should be added somewhere else

File: doc/deployment/README.md [hunk 8]
```diff
@@ -49,9 +68,17 @@ First, it's necessary to start some node/peer services:
 docker run \
     --network host \
     emcmongoose/mongoose[-<TYPE>] \
-    --run-node [\
-    --load-step-node-port=<PORT>]
+    --run-node 
 ```
+> NOTE: Mongoose uses `1099` port for RMI between mongoose nodes and `9999` for REST API. If you run several mongoose nodes on the same host (in different docker containers, for example) or if the ports are used by another service, then ports can be redefined:
```
Comment (dadlex) [inline doc/deployment/README.md]: Consider adding a simple example with 1 additional node and 1 master node showing usage of different ports

File: doc/design/architecture/README.md [hunk 9]
```diff
@@ -65,17 +71,27 @@
...
+Storage driver executes the *load operation*s generated by *Load Generator*s. It translates the Mongoose's abstract operations into the actual I/O requests and executes them. The basic property is the concurrency level and storage
 client configuration. The functionality includes:
 
 * Low-level implementation of the operations execution functionality
 * Rate limit related things
 * Callbacks for the completed operations
 
+![Storage drivers](../../images/storage_drivers.png)
+
+The driver is a "layer" between the *Load generator* and the tested Storage. Since storages of different types have different interaction protocols, then the implementation of Storage driver depends on the storage type. Therefore, the base image of the mongoose has only the implementation of the **Mock Storage driver** which does nothing actually and useful only for **demo/testing** purposes.
```
Comment (dadlex) [inline doc/design/architecture/README.md]: Therefore, the base image of the mongoose only has the implementation

File: doc/design/modes/README.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+![Standalone Mode](../../images/standalone_mode.png)
+
+This is the easiest way to use mongoose. In this case, one mongoose instance is launched and the load on the storage generated from it. This mode **does not require any additional parameters**.
+
+## Distributed mode
+
+![Distributed Mode](../../images/distributed_mode.png)
+
+This mod is used for distributed load on the storage. This mode is used **not** for a **distributed storage** (this function is also supported in a standalone mode), but for **several instances of the mongoose**.
+This mod is used to increase the load on the stoarge.
```
Comment (dadlex) [inline doc/design/modes/README.md]: storage

File: doc/design/modes/README.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+Mongoose can be launched in 2 main modes:
+* **Standalone** mode
+* **Distributed** mode
+
+## Standalone mode
+
+![Standalone Mode](../../images/standalone_mode.png)
+
+This is the easiest way to use mongoose. In this case, one mongoose instance is launched and the load on the storage generated from it. This mode **does not require any additional parameters**.
```
Comment (dadlex) [inline doc/design/modes/README.md]: This mode **does not require any additional parameters** - this is a bit unclear, what do you mean by that? Standalone mode might need a lot of parameters depending on a specific storage. I guess what you wanted to say is that it doesn't need additional parameters compared to additional parameters required for distributed mode, right?

File: doc/usage/load/operations/types/README.md [hunk 12]
```diff
@@ -79,7 +79,7 @@ items have been updated (modified).
 java -jar mongoose-<VERSION>.jar --read --item-data-verify --item-input-file=items.csv ...
 ```
 
-An load operation is reported with the [status code](../../../../interfaces/output#232-files) 12 if the content verification fails.
+An load operation is reported with the [status code](../../../output#232-files) 12 if the content verification fails.
```
Comment (dadlex) [inline doc/usage/load/operations/types/README.md]: A load operation

File: doc/deployment/README.md [hunk 13]
```diff
@@ -101,7 +128,7 @@ and run:
 docker run \
     --network host \
     emcmongoose/<IMAGE>:debug \
-    [<ARGS>]
+    [<MONGOOSE CLI ARGS>]
 ```
 
 # Kubernetes
```
Comment (dadlex) [inline doc/deployment/README.md:161]: line 140 - Helm not Melm

Comment (dadlex) [reply doc/deployment/README.md:161]: line 143 - if you move manifests to other dir, don't forget to change here

File: doc/getstarted/README.md (hunk 4 above)
Comment (veronikaKochugova) [reply doc/getstarted/README.md:1]: Load operations defenition? https://github.com/emc-mongoose/mongoose-base/tree/BASE-1389-documentation/doc/design/architecture#1-basic-terms

File: doc/deployment/README.md (hunk 13 above)
Comment (dadlex) [reply doc/deployment/README.md:161]: line 137 - consider stating explicitly: the only officially supported way to deploy Mongoose in Kubernetes is through Helm charts.

**Review by dadlex — CHANGES_REQUESTED** [review]

File: doc/getstarted/README.md (hunk 4 above)
Comment (dadlex) [reply doc/getstarted/README.md:1]: It says nothing about CRUD. so the connection between "write" and "create" isn't obvious

File: doc/deployment/README.md (hunk 8 above)
Comment (veronikaKochugova) [reply doc/deployment/README.md]: https://github.com/emc-mongoose/mongoose-base/blob/BASE-1389-documentation/doc/deployment/README.md#custom-ports

File: doc/design/modes/README.md (hunk 11 above)
Comment (veronikaKochugova) [reply doc/design/modes/README.md]: you are right

Comment (veronikaKochugova) [reply doc/design/modes/README.md]: https://github.com/emc-mongoose/mongoose-base/blob/BASE-1389-documentation/doc/design/modes/README.md#standalone-mode

File: doc/design/modes/README.md [hunk 14]
```diff
@@ -15,7 +15,7 @@ This is the easiest way to use mongoose. In this case, one mongoose instance is
 ![Distributed Mode](../../images/distributed_mode.png)
 
 This mod is used for distributed load on the storage. This mode is used **not** for a **distributed storage** (this function is also supported in a standalone mode), but for **several instances of the mongoose**.
-This mod is used to increase the load on the stoarge. 
+This mod is used to increase the load on the storage.
```
Comment (dadlex) [inline doc/design/modes/README.md]: mode

**Review by dadlex — CHANGES_REQUESTED** [review]

**Review by dadlex — APPROVED** [review]

**Pull request merged**